recursion/recursion.md

```diff
@@ -123,8 +123,72 @@ int fun1(int x, int y)
 ```
 
 What do these function calls return?
-
 `fun1(1, 1)`
 `fun1(2, 1)`
 `fun1(2, 2)`
 `fun1(0, 2)`
+`fun1(0, 2)`
+
+
+#### [Challenge] - GCD
+
+The [GCD](https://en.wikipedia.org/wiki/Greatest_common_divisor) of two or more integers, which are not all zero, is the largest positive integer that divides each of the integers.
+The greatest common divisor(GCD) is also known as:
+* the greatest common factor (GCF),
+* highest common factor (HCF),
+* greatest common measure (GCM),
+* highest common divisor.
+
+
+For example:
+
+the GCD of 48 and 14 is 2.
+
+### Pseudocode to calculate GCD of two numbers:
+
+```
+GCD(x, y)
+Begin
+      if y = 0 then
+          return x;
+      else
+          Call: GCD(y, x%y);
+      endif
+End
+```
+
+### Check for understanding
```

Comment:
This section is not clear to me at all. I’m not sure it’s necessary. It seems better to just ask learners to solve the problem with your pseudo code in the challenge section.